### ISSUE TITLE
Logically and Systematically Chaining PRs between Repositories

### ISSUE THREAD

**Issue by drparadoxical** [drparadoxical/aeon-rebase#14]
In order to keep ourselves on track while satisfying guidelines set forth by Smooth recently, I propose brute-forcing through the following so we can join the systems we're using together with the project management plan and requirements specification. 

1. Identify and add milestones to Github project space that reflect overarching work, such as instantiating local development environments, the operating environment for this project's testing and deployment of commits or reviewed and approved PRs, and standard operating procedure
   * This milestone was created just now to serve as an example for us here in this discussion
2. Identify and add Issues that are generic enough to allow the completion of its work to warrant a distinct PR, yet specific enough to allow work to be completed quickly (ill-defined milestone posing as an issue, for example)
3. For every PR reviewed and approved for our work here against drparadoxical/aeon-rebase, it will subsequently be issued as a new PR against aeonix/aeon-rebase.
   * i.e. approach issue->conduct work->issue PR to drparadoxical/aeon-rebase->team review of PR->consensus on approval or acknowledgment of clean review->issue PR to aeonix/aeon-rebase with reviewed/vetted/cleaned work

Ripping order from chaos, the SPMP (project management plan) brings the milestones and SOP into our shared reality and as a part of this system (Github Milestones), the SRS (requirements) brings the compartmentalized work into this system and grouped by aforementioned milestone. 

---

To put it bluntly, succinctly, and in the interest of bringing us and this work to center so we may move forward, this is how I propose we shove the IEEE docs for the project management plan and requirements specification into our systems, and work against those requirements as issues in milestones in a manner and method that gets it done here, gets it done there, all in accordance to all guidelines set forth. 

We wrap up the groundbreaking related issues as I complete the SPMP and SRS, we review and discuss those two PDFs, it turns into a couple hours worth of work to translate from consensus on approved SPMP and SRS docs to milestones and issues and well-defined SOP - and this is how Guidelines/Standards effectively collapse complex projects into systems and simple tasks. 

---

Opening this up for discussion now so we can prepare for the actual dropping into of this SOP once the OE is instantiated and we've closed the loop on pushing, deploying and testing, committing then reviewing PR here then issuing the final PR there, lather, rinse, repeat.

**Comment by guzzijones** [comment]
I like that each PR here will be a PR to Smooth's repo.  That will ensure each PR results in functional software.

**Comment by drparadoxical** [comment]
That's what I'm thinking, and we should be able to see how best to chain the PRs once the larger building blocks of the milestones/issues are in front of us here shortly (as the first pass, that is, no doubt I'll forgot something or get something wrong in the plan and spec).

**Comment by guzzijones** [comment]
FYi to do a PR to smooth's repo you will need to fork that repo .  We want to consider transitioning to that created repository instead of this one to make PR process simpler. 
Not anyones fault in fact i think the creation of this repo spawned Smooth's repo.  So we were a bit ahead of the curve.

**Comment by routed27** [comment]
Thanks for stating this. I was thinking this was the case but wasn't fully sure.

**Comment by devopsralf** [comment]
I think @guzzijones  you are correct as well. Also Merry Christmas everyone!

**Comment by guzzijones** [comment]
@drparadoxical  go ahead and fork the smooth repo and lets switch over to that one.

**Comment by routed27** [comment]
@guzzijones Is the plan, therefor, for @drparadoxical to fork the smooth repo, we then fork drparadoxicals version for our work. PRs will go against Dr's version (we can all review, etc). Eventually, drparadoxicals repo will submit PRs against smooths.
 Please correct me as needed.

**Comment by devopsralf** [comment]
That's how I understand it. @routed27

**Comment by guzzijones** [comment]
Yes. So we will have to move these milestones oner to that fork as well.

**Comment by drparadoxical** [comment]
Unless I'm mistaken, what we're working against right now is a clone serving as a stand-alone fork of the Smooth repo, in which we'll be forking ourselves to work against, all to PR against drparadoxical/aeon-rebase in order to finalize and subsequenly PR against aeonix/aeon-rebase. 

When all is said and done, we have a clean development and testing cycle, and a clean PR volley from a contributors work all the way into aeonix/aeon-rebase in the form of a reviewed/vetted/complete PR. 

If it isn't clear, this is essentially how I'm keeping everyone out of their own way, so that work can be done and go directly towards the rebase, without the rebase being a single, giant PR dump into aeonix/aeon-rebase when we're finished here. More importantly, this was one of Smooth's guidelines, in that we have to iteratively be pushing PRs based on our work, which will also keep us from doing work someone else may do outside of this group. 

In the event someone issues a PR that covers work we may have been doing here or intended to cover as a part of our plan, and assuming it's been approved on the aeonix/aeon-rebase side, we simply pull it down into our own repo here and keep on moving. 

The target we're trying to hit here, with this seemingly complex SOP, is create a space for us to develop and test and PR internally (drparadoxical/aeon-rebase) then PR externally (aeonix/aeon-rebase) in a manner and method that takes outside contributor PR work into consideration (contributor X, not a part of our team, has a PR approved and merged on aeonix/aeon-rebase, that we merge into drparadoxical/aeon-rebase accordingly, removing redundant effort and ensuring we waste as little time as possible here with this work). 

If it isn't clear to everyone, allow me to make it clear now - drparadoxical/aeon-rebase will be shuttered once the work is done, it is a temporary space meant to serve a single purpose: allow a rebase team to exist in order to contribute to rebasing aeonix/aeon to and through aeonix/aeon-rebase.

**Comment by guzzijones** [comment]
@drparadoxical we cannot do a PR against smooth's repo without first forking it. PR only exist on github not within git.

yes this is a clone, but not a fork.

**Comment by guzzijones** [comment]
either way.  you're going to have to fork Smooths' repo in order to do a PR.  So may as well move over to that fork once you do that.

**Comment by guzzijones** [comment]
go here: https://github.com/aeonix/aeon-rebase.  click fork.

**Comment by drparadoxical** [comment]
@guzzijones - we're in agreement, I'm assuming you haven't seen the note left in the project planning Gitter this morning: https://gitter.im/aeon-project-planning/Lobby

I've got to clear some assumptions and architecture/design decisions up, run them by Smooth, before I can finish the SPMP and SRS, but it's well understood our PR chaining methodology won't work unless we reset our starting point to be a direct fork of aeonix/aeon-rebase. The surrounding CI flow and optics of it all is a large part of why I want to get assumptions cleared up and ensure this is correct before we start in on the work and begin issuing these PRs up the ladder.

**Comment by drparadoxical** [comment]
But listen, two people catching the fly in the ointment early on is better than neither of us seeing it after getting a week of work into it all, so I appreciate you triple-pointing it out.

**Comment by guzzijones** [comment]
THANKS! nope haven't hit up gitter yet.  we are doing this right!

**Comment by drparadoxical** [comment]
Indeed. What's fair is fair, and what's right is right. 

Tautological, sure, but true is true nonetheless.

**Comment by guzzijones** [comment]
Just fyi, I don't want to start anything until we have everything issued out and people pick stuff. Otherwise we will end up working on the same thing and duplicating efforts.  Once we have milestone 2 worked out things should flow.  You geting the build bot up was awesome.

**Comment by drparadoxical** [comment]
Covered in the PMP overview and logistics. A bit looser than what you suggested, because there's really no stopping anyone from doing anything they'd like, but the gist in the form of reasonable guidelines that can easily be honored without hindering effort is as you've stated, more or less. 

The milestones and their respective issues, relative to decisions I believe should be made now instead of blindly copying functionality from the tag and the Monero team's decisions, is what I'm vetting and discussing now. Once I have feedback and can confirm, reject, or adjust my assumptions, then I can finalize the SRS, which brings the last bit of the SPMP together, which goes public. 

Then we review, discuss, tweak, educate, part out, and not delegate, but self-delegate and at least announce intent. Again, as you've said, just with the optics worked out beforehand, is all. 

Perfect on paper first, work is likely done when done once.

**Comment by drparadoxical** [comment]
This issue was moved to drparadoxical/aeon-rebase-1#6
